### DISCUSSION TITLE
[Feature] Smart Search: Advanced Search Filter to include/exclude Upload Library & External Library

### DISCUSSION THREAD

**Discussion by v3DJG6GL** [immich-app/immich#11975] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

The _Advanced search filters_ are already very helpful in many situations.
But what I still miss is an option to include and exclude assets that are stored in different libraries.
Background: I use the _Mobile Backup_ function to automatically back up the pictures I take with my mobile phone. However, I later copy them from my phone to a separate directory on my server as I edit these images manually with Lightroom.
This directory is mounted as an external library on my Immich instance. As a result, many of my images are now duplicated.
I can of course manually select the images that are stored in my upload library and manually delete each of these images. But a way to filter all images in my upload library would be so much easier :)

I think there are many other possible use cases where such a filter option would be very helpful. And since this information is already available for each asset in the database, I think it wouldn't be that difficult to add such an option. But that's just my guess, of course.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#7587/#5687
